modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/search/NodeSearchServiceImpl.java

```diff
@@ -51,7 +52,10 @@ private SearchResult doQuery( final NodeQuery query, final ReturnFields returnFi
         final SearchRequest searchRequest = SearchRequest.create().
             searchSource( source ).
             query( query ).
-            returnFields( returnFields ).
+            returnFields( query.isWithPath() ? ReturnFields.create().
```

Comment:
I don't currently imagine any other fields but `path` to be requested. Maybe we should deprecate/remove the method and move building of `ReturnFields` higher up

Reply:
there is also a request for REFERENCE filed. we need to keep it as it is for now